components/layout_2020/taffy/layout.rs

```diff
@@ -0,0 +1,607 @@
+    layout_context: &'a LayoutContext<'a>,
+    positioning_context: &'a mut PositioningContext,
+    content_box_size_override: &'a ContainingBlock<'a>,
+    style: &'a ComputedValues,
+}
+
+struct ChildIter(std::ops::Range<usize>);
+impl Iterator for ChildIter {
+    type Item = taffy::NodeId;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next().map(taffy::NodeId::from)
+    }
+}
+
+impl taffy::TraversePartialTree for TaffyContainerContext<'_> {
+    type ChildIter<'a> = ChildIter where Self: 'a;
+
+    fn child_ids(&self, _node_id: taffy::NodeId) -> Self::ChildIter<'_> {
+        ChildIter(0..self.source_child_nodes.len())
+    }
+
+    fn child_count(&self, _node_id: taffy::NodeId) -> usize {
+        self.source_child_nodes.len()
+    }
+
+    fn get_child_id(&self, _node_id: taffy::NodeId, index: usize) -> taffy::NodeId {
+        taffy::NodeId::from(index)
+    }
+}
+
+impl taffy::LayoutPartialTree for TaffyContainerContext<'_> {
+    type CoreContainerStyle<'a> = TaffyStyloStyle<&'a ComputedValues> where Self: 'a;
+    type CacheMut<'b> = AtomicRefMut<'b, taffy::Cache> where Self: 'b;
+
+    fn get_core_container_style(&self, _node_id: taffy::NodeId) -> Self::CoreContainerStyle<'_> {
+        TaffyStyloStyle(self.style)
+    }
+
+    fn set_unrounded_layout(&mut self, node_id: taffy::NodeId, layout: &taffy::Layout) {
+        let id = usize::from(node_id);
+        (*self.source_child_nodes[id]).borrow_mut().taffy_layout = *layout;
+    }
+
+    fn get_cache_mut(&mut self, node_id: taffy::NodeId) -> AtomicRefMut<'_, taffy::Cache> {
+        let id = usize::from(node_id);
+        let mut_ref: AtomicRefMut<'_, _> = (*self.source_child_nodes[id]).borrow_mut();
+        AtomicRefMut::map(mut_ref, |node| &mut node.taffy_layout_cache)
+    }
+
+    fn compute_child_layout(
+        &mut self,
+        node_id: taffy::NodeId,
+        inputs: taffy::LayoutInput,
+    ) -> taffy::LayoutOutput {
+        let mut child = (*self.source_child_nodes[usize::from(node_id)]).borrow_mut();
+        let child = &mut *child;
+
+        fn option_f32_to_lpa(input: Option<f32>) -> LengthPercentageOrAuto<Au> {
+            match input {
+                None => LengthPercentageOrAuto::Auto,
+                Some(length) => LengthPercentageOrAuto::LengthPercentage(Au::from_f32_px(length)),
+            }
+        }
+
+        fn option_f32_to_size(input: Option<f32>) -> Size<Au> {
+            match input {
+                None => Size::Initial,
+                Some(length) => Size::Numeric(Au::from_f32_px(length)),
+            }
+        }
+
+        with_independant_formatting_context(
+            &mut child.taffy_level_box,
+            |independent_context| -> taffy::LayoutOutput {
+                match independent_context {
+                    IndependentFormattingContext::Replaced(replaced) => {
+                        // The containing block of a grid item is the content box of the grid container
```

Comment:
The containing block is actually the grid area, see https://drafts.csswg.org/css-grid/#grid-item-sizing

Reply:
I believe this was true of `layout_2020`'s use of variables called "containing_block" (which seemed to mostly not actually be a containing block as the spec defines it) when this comment was written. But it looks like a lot of that has been cleaned up now. It's on my list to go through and re-evaluate the sizing constraint conversion logic with the latest layout_2020 changes.

Reply:
OK the provided `containing_block` may be wrong. But the comment seemed to refer to the containing block as defined in the spec.